tests/acceptance/core/testcase/localpathstorage.go

```diff
@@ -0,0 +1,80 @@
+package testcase
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/k3s-io/k3s/tests/acceptance/core/service/assert"
+	"github.com/k3s-io/k3s/tests/acceptance/shared/util"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+func TestLocalPathProvisionerStorage(g ginkgo.GinkgoTestingT, deployWorkload bool) {
+	if deployWorkload {
+		_, err := util.ManageWorkload(
+			"create",
+			"local-path-provisioner.yaml",
+			*util.Arch,
+		)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+			"local-path-provisioner manifest not deployed")
+	}
+
+	err := assert.ValidateOnHost(
+		util.GetPodVolumeTestRunning+util.KubeConfigFile,
+		util.RunningAssert,
+	)
+	if err != nil {
+		ginkgo.GinkgoT().Logf("Error: %v", err)
+	}
+
+	_, err = util.WriteDataPod(util.VolumeTest)
+	if err != nil {
+		_ = fmt.Errorf("error writing data to pod: %v", err)
+		return
+	}
+
+	gomega.Eventually(func(g gomega.Gomega) {
+		fmt.Println("Write and reading data from pod")
+		res, err := util.ReadDataPod(util.VolumeTest)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(res).Should(gomega.ContainSubstring(util.TestingLocalPath))
```

Comment:
This is a tricky thing about gomega and easy to miss. You have to use the `g gomega.Gomega` varaible for all checks within the Eventaually block. See [docs section](https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode) for more info.
```suggestion
		g.Expect(err).NotTo(gomega.HaveOccurred())
		g.Expect(res).Should(gomega.ContainSubstring(util.TestingLocalPath))
```